R/IDateTime.R

```diff
@@ -240,6 +240,30 @@ rep.ITime = function (x, ...)
   class(y) = "ITime"   # unlass and rep could feasibly not copy, hence use class<- not setattr()
   y
 }
+                           
+round.ITime <- function(x, units = c("hours", "minutes")) 
+{
+  (setattr(switch(match.arg(digits),
+                  hours = as.integer(round(unclass(x)/3600)*3600),
+                  minutes = as.integer(round(unclass(x)/60)*60)), 
+           "class", "ITime"))
+} 
+
+trunc.ITime <- function(x, units = c("hours", "minutes")) 
+{
+  (setattr(switch(match.arg(units),
+                  hours = as.integer(trunc(unclass(x)/3600)*3600),
+                  minutes = as.integer(trunc(unclass(x)/60)*60)), 
+           "class", "ITime"))
+}
+
+floor.ITime <- function(x) {
```

Comment:
why should we expect `floor(ITime)` to round to the nearest hour? do you have a use case for this function?

Reply:
At least, why shouldn't the signature here have `units` as well?

Reply:
`floor` does not allow more than one argument. I considered making two functions: `floor_hour` and `floor_minute`, but decided against it.

Reply:
OK, understood. Is there any reason to believe `hour` is the more common use case?

Reply:
No not really.

Perhaps these should just be removed so that the decision of rounding will be up to the user.

Reply:
Are you sure you're not allowed to have a `units` argument?

Reply:
> floor.ITime <- function(x, units = c("hours", "minutes")) {
+     (setattr(switch(match.arg(units),
+                     hours = as.integer(floor(unclass(x)/3600L)*3600L),
+                     minutes = as.integer(floor(unclass(x)/60L)*60L)), 
+              "class", "ITime"))
+ }
> 
> floor(DT$time, units = "min")

Error in floor(DT$time, units = "min") : 
  2 arguments passed to 'floor' which requires 1

Reply:
Yep :)

![image](https://user-images.githubusercontent.com/37183160/73530047-06ab1280-4418-11ea-8a38-830f62892b97.png)